data_acquisition/src/general_functions.R

```diff
@@ -0,0 +1,138 @@
+                       name <- names(nested$param[[x]])
+                       nested$desc[x] <- name
+                       nested <- nested %>% 
+                         unnest(param) %>% 
+                         mutate(param = as.character(param))
+                       nested[x,]
+                       })
+  # re-orient to make it easy to grab necessary info in future functions
+  unnested <- unnested %>% 
+    select(desc, param) %>% 
+    pivot_wider(names_from = desc, values_from = param)
+  write_csv(unnested, 'data_acquisition/in/yml.csv')
+  'data_acquisition/in/yml.csv'
+}
+
+
+# grabLocs: Load in and format location file using yaml file ----
+
+grabLocs <- function(yaml) {
+  locs <- read_csv(file.path(yaml$data_dir, yaml$location_file))
+  # store yaml info as objects
+  lat <- yaml$latitude
+  lon <- yaml$longitude
+  id <- yaml$unique_id
+  # apply objects to tibble
+  locs <- locs %>% 
+    rename_with(~c('Latitude', 'Longitude', 'id'), any_of(c(lat, lon, id)))
+  write_csv(locs, 'data_acquisition/in/locs.csv')
+  'data_acquisition/in/locs.csv'
+}
+
+
+# getNHD: if user desires lake extent and does not provide lake polygons, use NHDPlus to grab and export polygons
+
+getNHD <- function(locs, yaml) {
+  # read in files
+  locations = read_csv(locs)
+  yaml = read_csv(yaml)
+  if (grepl('poly', yaml$extent[1])) { # if polygon is specified in desired extent - either polycenter or polgon
+    # create sf
+    wbd_pts = st_as_sf(locations, crs = yaml$location_crs[1], coords = c('Longitude', 'Latitude'))
+    id = locations$id
+    
+    if (yaml$polygon[1] == 'FALSE') { # and no polygon is provided, then use nhdplustools
+      for(w in 1:length(id)) {
+        aoi_name = wbd_pts[wbd_pts$id == id[w],]
+        lake = get_waterbodies(AOI = aoi_name)
+        if (w == 1) {
+          all_lakes = lake
+        } else {
+          all_lakes = rbind(all_lakes, lake)
+        }
+      }
+      all_lakes = all_lakes %>% select(id, comid, gnis_id:elevation, meandepth:maxdepth)
+      write_csv(st_drop_geometry(all_lakes), 'data_acquisition/out/NHDPlus_stats_lakes.csv')
+      all_lakes = all_lakes %>% select(id, comid, gnis_name)
+    } else { # otherwise read in specified file
+      all_lakes = read_sf(file.path(yaml$lake_poly_dir[1], yaml$lake_poly_file[1])) 
+      write_csv(st_drop_geometry(all_lakes) %>% rowid_to_column('id'), 'data_acquisition/out/user_lakes_withrowid.csv')
+    }
+    st_write(all_lakes, 'data_acquisition/out/lakes.shp')
+    return('data_acquisition/out/lakes.shp')
+  } else {
+    return(print('Not configured to use polygon area.'))
+  }
+}
+
+
+# calcCenter
+calcCenter <- function(poly, yaml) {
+  yaml = read_csv(yaml)
+  if (grepl('center', yaml$extent[1])) {
+    # load polygon
+    polygon = read_sf(poly)
+    # create an empty tibble
+    cc_df = tibble(
+      rowid = integer(),
+      lon = numeric(),
+      lat = numeric(),
+      dist = numeric()
+    )
+    for (i in 1:length(polygon[[1]])) {
```

Comment:
Could this be a `map()` statement instead of a loop? Personal preference more than anything